### PR TITLE
[UPD] add new functions, correcting docstring and add more examples

### PR DESCRIPTION
**Pull request by nhatnm0612**
## Summary

Add new functions, correct docstrings.

## Details

- [x] getServicePoint(name) -> getServicePoint(id)
- [x] searchServicePoint(domain)
- [x] getArea(id)
- [x] searchArea(domain)
- [x] correct docstring for old methods
- [x] create docstring with multi examples

## Issues

[link to issue](https://github.com/OpenSPP/openspp-registry/issues/248#issuecomment-1827025997)

## Review Checklist

Before merging, the reviewer should check the following items:

- [ ] Does the PR do what it claims to do?
- [ ] If this is a new adaptor, has the migration tool been run and the
      migration guide followed?
- [ ] Are there any unit tests? Should there be?
- [ ] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.

### CODE REVIEW

**Review by jeremi — CHANGES_REQUESTED** [review]
can you review the parameters to those 2 functions as commented?

File: packages/openspp/src/Adaptor.js [hunk 1]
```diff
@@ -918,6 +920,164 @@ export function removeFromGroup(group_id, individual_id) {
...
+ * @function
+ * @param {Array} domain - searching domain
+ * @param {number} [offset=0] - offset searching
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function searchArea(domain, offset=0, callback=(s) => s) {
+  return async state => {
+    try {
+      let isDomain = true;
```
Comment (josephjclark) [inline packages/openspp/src/Adaptor.js:1040]: This block of code, which scans `domain` and maybe embeds it in an array, is repeated. It would be nice to factor it out into another function and just call it from here.
```
const resolveDomain = (domain) => {
       let isDomain = true;
      for (const element of domain) {
        if (!Array.isArray(element)) {
          isDomain = false;
          break;
        }
      }
      if (!isDomain) {
        domain = [domain];
      }
     return domain;
}

// then in your operation:
```
try {
   domain = resolveDomain(domain)
    const areas = await sppConnector.searchRead( ... )
}
```
I think this does the same thing but a bit more cleanly?
```
const resolveDomain = (domain) => {
    for (const element of domain) {
        if (!Array.isArray(element)) {
            return [domain]
        }
    }
}
```
We normally put functions like this into a `Utils.js` file but that's not strictly necessary. Also, functions like this are great to unit test!

**Review by josephjclark — COMMENTED** [review]
Hi, just a few comments from me.

Let us know when you're ready and we'll get this merged and released for you.

File: packages/openspp/src/Adaptor.js [hunk 2]
```diff
@@ -918,6 +920,164 @@ export function removeFromGroup(group_id, individual_id) {
...
+ * searchServicePoint([["name", "ilike", "agent 1"]])
+ * @example <caption>search with offset</caption>
+ * searchServicePoint([["name", "ilike", "agent 1"]], 3)
+ * @function
+ * @param {Array} domain - searching domain
+ * @param {number} [offset=0] - offset searching
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function searchServicePoint(domain, offset=0, callback=(s) => s) {
```
[the remark this answers is not in the document]
Comment (nhatnm0612) [reply packages/openspp/src/Adaptor.js:936]: I did this intentionally, then sadly it's a no, til someone on our company wants me to do so. It's limiting query in specific order, and I didn't want to give all data back via api

File: packages/openspp/src/Adaptor.js [hunk 3]
```diff
@@ -57,27 +57,43 @@ async function login(state) {
   return state;
 }
 
+/**
+ * resolve input domain
```
Comment (josephjclark) [inline packages/openspp/src/Adaptor.js:61]: I appreciate the documentation on this but just for future reference, you don't need to add JSDocs to internal functions like this :) You only need to JSDoc the exported public functions - the stuff users will use in job code.

Don't remove this, it's great, just bear it in mind for next time :)

File: packages/openspp/src/Adaptor.js [hunk 4]
```diff
@@ -918,6 +918,146 @@ export function removeFromGroup(group_id, individual_id) {
...
+ * searchArea([["code", "=", "10732"]])
+ * @example <caption>search with offset</caption>
+ * searchArea([["kind", "=", 1]], 1)
+ * @function
+ * @param {Array} domain - searching domain
+ * @param {number} [offset=0] - offset searching
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function searchArea(domain, offset=0, callback=(s) => s) {
```
Comment (josephjclark) [inline packages/openspp/src/Adaptor.js:1026]: Oh, this should take an `options` object too!

Comment (nhatnm0612) [reply packages/openspp/src/Adaptor.js]: @josephjclark done 😄

File: packages/openspp/src/Adaptor.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nhatnm0612) [reply packages/openspp/src/Adaptor.js]: done 😄

File: packages/openspp/ast.json [hunk 5]
```diff
@@ -190,7 +189,7 @@
           },
           {
             "title": "example",
-            "description": "getServicePoint(\"000117\")"
+            "description": "getServicePoint(1)"
```
[the remark this answers is not in the document]
Comment (nhatnm0612) [reply packages/openspp/ast.json]: @jeremi done changing 😄

File: packages/openspp/ast.json [hunk 6]
```diff
@@ -954,6 +973,205 @@
...
+        "description": "get area by id in OpenSPP",
+        "tags": [
+          {
+            "title": "public",
+            "description": null,
+            "type": null
+          },
+          {
+            "title": "example",
+            "description": "getArea(1)"
```
[the remark this answers is not in the document]
Comment (nhatnm0612) [reply packages/openspp/ast.json]: done changing 😄

**Review by josephjclark — APPROVED** [review]

**Comment by josephjclark** [comment]
@mtuchi Can I ask you to release from this branch today and then merge this into main? Thank you! :pray:

File: packages/openspp/CHANGELOG.md
Comment (mtuchi) [inline packages/openspp/CHANGELOG.md:1]: @josephjclark why is this added manually and not changeset ?

**Comment by josephjclark** [comment]
Hi @nhatnm0612, `@openfn/language-openspp` 1.2.0 is now live!
